src/main/java/ch/njol/skript/effects/EffPotion.java

```diff
@@ -36,143 +36,129 @@
 import ch.njol.skript.util.Timespan;
 import ch.njol.util.Kleenean;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Potion Effects")
 @Description("Apply or remove potion effects to/from entities.")
-@Examples({"apply swiftness 2 to the player",
+@Examples({
+	"apply ambient swiftness 2 to the player",
 	"remove haste from the victim",
+	"",
 	"on join:",
-	"\tapply potion of strength of tier {strength.%player%} to the player for 999 days",
-	"apply potion effects of player's tool to player"})
-@Since("2.0, 2.2-dev27 (ambient and particle-less potion effects), 2.5 (replacing existing effect), 2.5.2 (potion effects)")
+		"\tapply infinite potion of strength of tier {strength::%uuid of player%} to the player",
+		"\tapply potion of strength of tier {strength::%uuid of player%} to the player for 999 days # Before 1.19.4",
+	"",
+	"apply potion effects of player's tool to player",
+	"apply haste potion of tier 3 without any particles to the player whilst hiding the potion icon # Hide potions"
+})
+@Since(
+	"2.0, 2.2-dev27 (ambient and particle-less potion effects), " + 
+	"2.5 (replacing existing effect), 2.5.2 (potion effects), " +
+	"INSERT VERSION (icon and infinite)"
+)
 public class EffPotion extends Effect {
+
 	static {
 		Skript.registerEffect(EffPotion.class,
-			"apply %potioneffects% to %livingentities%",
-			"apply [potion of] %potioneffecttypes% [potion] [[[of] tier] %-number%] to %livingentities% [for %-timespan%] [(1¦replacing [the] existing effect)]",
-			"apply ambient [potion of] %potioneffecttypes% [potion] [[[of] tier] %-number%] to %livingentities% [for %-timespan%] [(1¦replacing [the] existing effect)]",
-			"apply [potion of] %potioneffecttypes% [potion] [[[of] tier] %-number%] without [any] particles to %livingentities% [for %-timespan%] [(1¦replacing [the] existing effect)]"
-			//, "apply %itemtypes% to %livingentities%"
-			/*,"remove %potioneffecttypes% from %livingentities%"*/);
+				"apply %potioneffects% to %livingentities%",
+				"apply infinit(e|y) [:ambient] [potion of] %potioneffecttypes% [potion] [[[of] tier] %-number%] [particles:without [any] particles] to %livingentities% [replacing:replacing [the] existing effect] [icon:hide [the] [potion] icon]",
+				"apply [:ambient] [potion of] %potioneffecttypes% [potion] [[[of] tier] %-number%] [particles:without [any] particles] to %livingentities% [for %-timespan%] [replacing:replacing [the] existing effect] [icon:[whilst] hid(e|ing) [the] [potion] icon]"
```

Comment:
Optional but I think this makes more sense otherwise it is misleading
```suggestion
				"apply [:ambient] [potion of] %potioneffecttypes% [potion] [[[of] tier] %-number%] [noparticles:without [any] particles] to %livingentities% [for %-timespan%] [replacing:replacing [the] existing effect] [icon:[whilst] hid(e|ing) [the] [potion] icon]"
```